csharp/src/Drivers/Apache/Thrift/BitmapUtilities.cs

```diff
@@ -0,0 +1,41 @@
+﻿/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using static Apache.Arrow.ArrowBuffer;
+
+namespace Apache.Arrow.Adbc.Drivers.Apache.Thrift
+{
+    internal static class BitmapUtilities
+    {
+        /// <summary>
+        /// Gets the "validity" bitmap buffer from a 'nulls' bitmap.
+        /// </summary>
+        /// <param name="nulls">The bitmap of rows where the value is a null value (i.e., "invalid")</param>
+        /// <param name="nullCount">Returns the number of bits set in the bitmap.</param>
+        /// <returns>A <see cref="ArrowBuffer"/> bitmap of "valid" rows (i.e., not null values).</returns>
+        /// <remarks>Inverts the bits in the incoming bitmap to reverse the null to valid indicators.</remarks>
+        internal static ArrowBuffer GetValidityBitmapBuffer(byte[] nulls, out int nullCount)
+        {
+            nullCount = BitUtility.CountBits(nulls);
```

Comment:
This is both incorrect and (probably) slow.

BitUtility.CountBits returns the number of bits that are set, but we presumably need to toggle all the bits in the range, whether or not they are set. The only way to do this correctly is to pass in the number of elements in the array. All of the callers should have this information.

A likely-faster approach is then to do bytewise flipping until we get for (count / 8) bytes and then bitwise flipping for the remaining (count % 8) bits. We can almost certainly do better by flipping the bits in larger chunks, but at the expense of more complicated logic.

Reply:
@CurtHagenlocher 
I've optimized this to do integer/byte/bit-wise flipping. Added a test to ensure that there is no byte-order or bit-order bugs.